torch/csrc/jit/script/logging.cpp

```diff
@@ -0,0 +1,85 @@
+#include "torch/csrc/jit/script/logging.h"
+
+#include <atomic>
+#include <mutex>
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+namespace logging {
+
+// TODO: multi-scale histogram for this thing
+
+void LockingLogger::addStatValue(const std::string& stat_name, int64_t val) {
+  std::unique_lock<std::mutex> lk(m);
+  raw_counters[stat_name].push_back(val);
+}
+
+TORCH_API int64_t LockingLogger::getCounterValue(const std::string& name) const {
+  const std::vector<int64_t> *vals;
+  int64_t retval;
+  AggregationType type;
+  {
+    std::unique_lock<std::mutex> lk(m);
+    if (!raw_counters.count(name)) {
+      return 0;
+    }
+    type = agg_types.count(name) ? agg_types.at(name)
+                                                : AggregationType::SUM;
+    vals = &raw_counters.at(name);
+  }
+  switch (type) {
```

Comment:
race! You released the lock on the hash table. It has resized and then moved the contained vector to another address. `*vals` is invalid.